src/DebuggerForm.cpp

```diff
@@ -1537,6 +1591,50 @@ void DebuggerForm::processMerge(const QString& message)
 	}
 }
 
+QByteArray DebuggerForm::saveCommands() const
+{
+	QByteArray output;
+
+	auto first = true;
+	for (const auto& [name, description, source, icon, _] : commands) {
+		if (!first) output.append(':');
+		first = false;
+		output.append(name.toUtf8().toBase64());
+		output.append(';');
+		output.append(description.toUtf8().toBase64());
+		output.append(';');
+		output.append(source.toUtf8().toBase64());
+		output.append(';');
+		output.append(icon.toUtf8().toBase64());
+	}
+
+	return output;
+}
+
+void DebuggerForm::restoreCommands(const QByteArray &input)
```

Comment:
Detail: in most other code we use the style `const QByteArray& input` (space on the other side of `&`).